lib/rubocop/cop/lint/useless_trailing_comma.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks whether trailing commas in block arguments are
+      # required. Blocks with only one argument and a trailing comma require
+      # that comma to be present. Blocks with more than one argument never
+      # require a trailing comma.
+      #
+      # @example
+      #   # bad
+      #   add { |foo, bar,| foo + bar }
+      #
+      #    # good
+      #   add { |foo, bar| foo + bar }
+      #
+      #   # good
+      #   add { |foo,| foo }
+      #
+      #   # good
+      #   add { foo }
+      #
+      #   # bad
+      #  add do |foo, bar,|
+      #    foo + bar
+      #   end
+      #
+      #  # good
+      #  add do |foo, bar|
+      #    foo + bar
+      #  end
+      #
+      #  # good
+      #   add do |foo,|
+      #    foo
+      #  end
+      #
+      #  # good
+      #  add do
+      #     foo + bar
+      #  end
+      class UselessTrailingComma < Cop
+        include RangeHelp
+
+        MSG = 'Block arguments do not require a trailing comma except when ' \
+              'only a single argument is present.'
+
+        ARG_MSG = 'Useless trailing comma present in block arguments.'
```

Comment:
If this is the message we want, we can assign this to `MSG` and not need to pass it around inside the implementation. (It is used by default by `#add_offense`.)